WordPress/gutenberg
# Add block appender to Nav block offcanvas experiment

## What?


Adds a basic appender to the Offcanvas Nav block experiment.

This does not fully implement the appender. Further work is required to customise the appender to show the UI in the sidebar rather than on the canvas.

Consider also comparing this with https://github.com/WordPress/gutenberg/pull/45947.

Part of https://github.com/WordPress/gutenberg/issues/45445.

Co-authored-by: Dave Smith <getdavemail@gmail.com> 

## Why?


Addresses the first part of https://github.com/WordPress/gutenberg/issues/45445.

## How?


Adds a custom appender to the list view sidebar.


## Testing Instructions
1. Enable offcanvas experiment.
2. New Post -> Add Nav block.
3. Open offcanvas
4. See appender.
5. Click appender.
6. See that it opens the inserter _in the canvas_ (that's intentional....**for now**).

## Screenshots or screencast 

https://user-images.githubusercontent.com/444434/203079007-5e6a7624-09e9-4e20-81be-89c17bd468af.mp4


<img width="1161" alt="Screen Shot 2022-11-21 at 14 24 25" src="https://user-images.githubusercontent.com/444434/203079000-4bab1546-9ded-444e-8ae3-0466ae3192e1.png">
